SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1423,202 +1393,140 @@ bool RunSchemaCompiler(FString& SchemaBundleJsonOutput, FString SchemaInputDir,
 	}
 }
 
-bool ExtractInformationFromSchemaJson(const FString& SchemaJsonPath, TMap<uint32, FComponentIDs>& OutComponentSetMap,
-									  TMap<uint32, uint32>& OutComponentIdToFieldIdsIndex, TArray<FFieldIDs>& OutFieldIdsArray)
+bool CreatePartitionAuthoritySet(FString SchemaInputPath, FString SchemaOutputPath)
 {
-	TUniquePtr<FArchive> SchemaFile(IFileManager::Get().CreateFileReader(*SchemaJsonPath));
-	if (!SchemaFile)
+	if (SchemaOutputPath == "")
 	{
-		UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Could not open schema bundle file %s"), *SchemaJsonPath);
-		return false;
+		SchemaOutputPath = GetDefault<USpatialGDKEditorSettings>()->GetGeneratedSchemaOutputFolder();
 	}
 
-	TSharedPtr<FJsonValue> SchemaBundleJson;
-	{
-		TSharedRef<TJsonReader<char>> JsonReader = TJsonReader<char>::Create(SchemaFile.Get());
-		FJsonSerializer::Deserialize(*JsonReader, SchemaBundleJson);
-	}
+	FString IntermediateDir = GenerateIntermediateDirectory();
+	const FString PartitionDataFolderName(TEXT("PartitionMetadata"));
 
-	const TSharedPtr<FJsonObject>* RootObject;
-	if (!SchemaBundleJson || !SchemaBundleJson->TryGetObject(RootObject))
+	if (SchemaInputPath == "")
 	{
-		UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("%s is not a valid Json file"), *SchemaJsonPath);
-		return false;
+		FString ContentDir = FPaths::ProjectContentDir();
```

Comment:
FPath::ProjectContentDir returns a temporary.

Reply:
Shouldn't be a problem - https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/

Reply:
I see. Still, if someone where to pass that reference somewhere else, I'd rather make a local copy to showcase the temporary nature of the var. Seeing this code, it would be unlikely for the dev to look at the function to determine if it's actually a temporary or a persistent reference.